hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -462,10 +478,23 @@ else if (methodName.startsWith("get") ||
 					}
 				}
 			}
-			if (propertyHasAnnotation && !propertyNameMatchesFieldName) {
-				log.debugf("Skipping enhancement of [%s]: due to class [%s] not having a property accessor method name matching field name [%s]",
-						managedCtClass, methodDescription.getDeclaringType().getActualName(), methodFieldName);
-				return true;
+			if ( propertyHasAnnotation && !propertyNameMatchesFieldName ) {
+				switch ( strategy ) {
+					case SKIP:
+						log.debugf(
+								"Skipping enhancement of [%s] because no field named [%s] could be found for property accessor method [%s]."
+										+ " To fix this, make sure all property accessor methods have a matching field.",
+								managedCtClass.getName(), methodFieldName, methodDescription.getName() );
+						return true;
```

Comment:
methodFieldName is basically the property name derived from the method name which might be derived from an actual method name like `getIdentifier()` for which the property name is `identifier`.  So it is derived from the `property accessor method`.  Hope that makes sense.

The methodDescription.getDeclaringType().getActualName() may be the same class as managedCtClass.getName() but may also be a super class.  When it is a super class, it might be helpful to mention that name so users understand that is where the property name is from.

Reply:
> methodFieldName is basically the property name derived from the method name which might be derived from an actual method name like `getIdentifier()` for which the property name is `identifier`. So it is derived from the `property accessor method`. Hope that makes sense.

It does, I think we're on the same page. I just didn't find a way to can convey that meaning in a short, easy-to-read, understandable message. Hence this wording.

Note I changed the log message just to align on the exception, which absolutely must be short and understandable without context -- but I can revert the debug log to what it was if you think it's best.

> The methodDescription.getDeclaringType().getActualName() may be the same class as managedCtClass.getName() but may also be a super class. When it is a super class, it might be helpful to mention that name so users understand that is where the property name is from.

I can add that name back if you feel strongly about it, but there too, I feared the longer the message, the harder it would be to read/understand.

Reply:
I could be wrong but I'm concerned that the user might not understand which classes in their application need to change if we only show the name of the class being enhanced but not the name of class that has the problem.  Of course this is only a problem if `methodDescription.getDeclaringType()` is not the same class as managedCtClass.

Reply:
I defer to what the ORM team prefers (e.g. shorter message versus longer message with more details that mention involved class names).

Reply:
It's a DEBUG-level message. No user is ever going to see it.

Reply:
(Perhaps you are talking about the exception thrown lower down.)